ledger-tool/src/main.rs

```diff
@@ -1190,6 +1253,14 @@ fn main() {
                     "Mode to recovery the ledger db write ahead log"
                 ),
         )
+        .arg(
+            Arg::with_name("force_create_if_missing")
+                .long("force-create-if-missing")
+                .takes_value(false)
+                .global(true)
+                .help("Allow commands that would otherwise not alter the ledger \
+                       to create file(s) necessary to open the blockstore"),
+        )
```

Comment:
I called the flag `force_create_if_missing`, what do you two think of the flag name + help string ?

Reply:
create_if_missing makes me think of the create the blockstore if it does not exist, while in the current context we are talking about creating a new column family if needed.  If it's the decision between `force_create_if_missing` and `force-open-blockstore`, `force-open-blockstore` sounds a little bit better.

Since we are looking for a flag that will grant us temporary write access to do enough option changes to open the db in different versions, how about something like `open-with-compatibility-mode` and document clearly that the ledger tool may change necessary options / add missing column families in order to open the blockstore?

Reply:
> create_if_missing makes me think of the create the blockstore if it does not exist, while in the current context we are talking about creating a new column family if needed

I actually expanded the check; it will create a missing blockstore or just missing columns. So, I was trying to make the name applicable for both, creating missing column(s) or create entirely missing blockstore.

> how about something like open-with-compatibility-mode and document clearly that the ledger tool may change necessary options / add missing column families in order to open the blockstore?

Thanks for the suggestion, `compatibility-mode` could be an option. Let me think about this in the background a little more; there isn't a huge rush to get this in so we can sit on it for a week and see if any new ideas hit. Maybe `--force-open-and-update` or `--force-update-to-open`, the latter implies that the update is necessary in order to open the blockstore.

> document clearly that the ledger tool may change necessary options / add missing column families in order to open the blockstore?

I think the help string associated with the flag is self-documenting in this way; did you see that / have any thoughts on that string ?

Reply:
> Let me think about this in the background a little more; there isn't a huge rush to get this in so we can sit on it for a week and see if any new ideas hit. Maybe --force-open-and-update or --force-update-to-open, the latter implies that the update is necessary in order to open the blockstore.

Feel free to let me know if you have additional thoughts or updates.  In general, the direction of this PR looks good to me.

Reply:
This one fell off my radar until stale-bot gave me the push too. In any case, I'm leaning towards `force-update-to-open`; you can see the updated help string in updated commit as well

- I liked `force-create-if-missing` initially; however, this is very specific to the current cases (missing column or missing db). Being more generic could allow us to re-use the flag for more similar cases down the road.
- I think `open-with-compatibility-mode` might imply that we're changing runtime software settings only (ie the mode that we're using to open) whereas we are actually updating the db on disk.

Reply:
Let's do `force-update-to-open`!